bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -502,9 +515,10 @@ private Rendition getAWebRendition(Asset asset) {
         return null;
     }
 
-    private void stream(@Nonnull SlingHttpServletResponse response, @Nonnull InputStream inputStream, @Nonnull String contentType)
+    private void stream(@Nonnull SlingHttpServletResponse response, @Nonnull InputStream inputStream, @Nonnull String contentType, String imageName)
             throws IOException {
         response.setContentType(contentType);
+        response.setHeader( "Content-Disposition", "filename=\"" + imageName + "\"" );
```

Comment:
Shouldn't we also include the file extension here?

Reply:
I have added it now. Couldn't find a reason for not including this.